controllers/provider-aws/pkg/validator/serialization.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validator
+
+import (
+	"github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/aws"
+	"github.com/gardener/gardener-extensions/pkg/util"
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func decodeWorkerConfig(decoder runtime.Decoder, worker *gardencorev1alpha1.Worker, fldPath *field.Path) (*aws.WorkerConfig, error) {
+	workerConfig := &aws.WorkerConfig{}
+	if err := util.Decode(decoder, worker.ProviderConfig.Raw, workerConfig); err != nil {
```

Comment:
What if `providerConfig` is `nil`? (same for others below)

Reply:
It's up to the caller to check that first because he can only know if that's a valid point or not. I changed the signature of this function to make it more clear.